.github/workflows/ci.yml

```diff
@@ -47,12 +47,13 @@ jobs:
 
   unitpreOS:
     strategy:
+      fail-fast: false
       matrix:
       # us-east-2
       # Ubuntu 20.04 AMI is ami-0e4d0bb9670ea8db0
       # ubuntu 22.04 AMI is ami-05fb0b8c1424f266b 
       # RHEL 9       AMI is ami-078cbc4c2d057c244
-        AMI_ID: [ami-0e4d0bb9670ea8db0,ami-05fb0b8c1424f266b,ami-078cbc4c2d057c244]
```

Comment:
@rootfs , it seems we have 2 rhel AMI id in the list, could you please check?

Reply:
these are 2 ubuntu and 1 rhel

Reply:
but from the test output,
https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/actions/runs/8234510994/job/22516556954?pr=18#step:3:12
https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/actions/runs/8234510994/job/22516559265?pr=18#step:3:11

it seems we have two redhat instance?

Reply:
Machine name: 'ip-172-31-7-40'
https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/actions/runs/8234510994/job/22516556954?pr=18#step:3:12
Machine name: 'ip-172-31-7-40'
https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/actions/runs/8234510994/job/22516559265?pr=18#step:3:11
it seems the two jobs been set into same instance.

Reply:
btw, @sunya-ch , may I know which ami id and instance configuration is used for model server for now?

Reply:
Please find the configuration below:
```
kepler_model_server_image: quay.io/sustainable_computing_io/kepler_model_server:v0.7
ami_id: ami-0e4d0bb9670ea8db0
region: us-east-2
secuirty_group: sg-cda39bb6
volume_size: 120
key_name: kepler-ci-key
instance_list:
- m5zn.metal 
- i3.metal 
- m5dn.metal
- m5n.metal
- c5d.metal 
- c7i.metal-24xl
- i3en.metal 
- m7i.metal-24xl 
```

reference: https://github.com/sustainable-computing-io/kepler-model-training-playbook/blob/main/group_vars/local.yml

Reply:
it seems t2.micro is too small to run the kind cluster. https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/actions/runs/8314627605/job/22752095043?pr=18#step:5:685

I don't want CI charges a lot.  @rootfs , @sunya-ch , any suggestion for a minimal size instance?

Reply:
You can check the spot price that @rootfs shared here: https://gist.github.com/rootfs/49855d0c75ef8a27e494eda6eb928aab

Also the spot instance I have tested with data collection here: https://github.com/sustainable-computing-io/kepler-model-training-playbook/blob/main/README.md

The cheapest one that can run seem to be `m5zn.metal`.

Reply:
@SamYuan1990  Agree. I don't think we need all profile for CI testing. Also, we need to limit the path to activate this CI only when necessary. The list above in playbook repo is for collecting the data for power model training. We need to cover as much as possible. But, we do collect the data only once if there is no impacting change on Kepler exporter such as metric name.